### ISSUE TITLE
loosing @default(cuid()) @id after running introspect

### ISSUE THREAD

**Issue by alan345** [prisma/upgrade#59]
The first time I run `npx prisma introspect` it recognizes it was a primsa1 database. So it added automatically `@default(cuid()) @id` 

```
model Companie {
  id                                    String                       @default(cuid()) @id
}
```
I run `npx prisma-upgrade` to update my database.

I run `npx prisma introspect` again after upgrading the dababase. now in the `schema.prisma` Ids become : `@id`
```
model Companie {
  id                                    String                       @id
}
```

Is there a way to keep this `cuid()` info after running introspect?

**Comment by janpio** [comment]
Can you share the relevant SQL from before and after you did what `prisma-upgrade` told you to apply to that table?

**Comment by alan345** [comment]
for the object `Balance` for example:
```
    SET FOREIGN_KEY_CHECKS=0;
    ALTER TABLE `Balance` CHANGE `id` `id` char(30) CHARACTER SET utf8 NOT NULL;
```

```
    UPDATE `Balance`, `_CompanieBalanceRelation` SET `Balance`.`companieId` = `_CompanieBalanceRelation`.B where `_CompanieBalanceRelation`.A = `Balance`.`id`;
    ALTER TABLE `Balance` CHANGE `companieId` `companieId` char(25) CHARACTER SET utf8 NOT NULL;
    ALTER TABLE `Balance` ADD FOREIGN KEY (`companieId`) REFERENCES `Companie`(`id`);
```

**Comment by janpio** [comment]
There _might_ be some special handling for `id + char(25)` in Introspection, that is not triggered any more after changing the column to `char(30)`. Can you please check that @do4gr? If so, we might want to add the 30 here as an alternative was this is what we are telling P1 users to change it to.

**Comment by alan345** [comment]
Hi guys. Any news? Let me know if you need more data/info from my side

**Comment by sktt** [comment]
I seen this too with postgres, after updating an id column to from varying(25) to varying(30) it switches to `@id` as described here. I thought this was intended behavior. Why does it make any difference? They'll be strings either way, no?

**Comment by janpio** [comment]
See https://github.com/prisma/upgrade/issues/59#issuecomment-669520099 @sktt

**Comment by do4gr** [comment]
@janpio 's suspicion is correct. There is special handling for the length of the char column in the database to reduce the likelihood of false positives when inferring the `cuid()`. That logic needs to be adjusted to also now include the new length of 30 introduced by the upgrade tool. I'll do it and this should soon be fixed.

**Comment by janpio** [comment]
Hey @alan345, while investigating this further we found 2 things:

1. What @do4gr posted above will probably not work unfortunately. The other changes of the upgrade tool already mean this special logic does not trigger any more - or so we think.
    You could help us to confirm this by running the upgrade tool suggestion but NOT the one that switches the cuid column from `varchar(25)` to `varchar(30)`. Does `prisma introspect` still recognize this field as `cuid()` after doing that?
2. We have a better solution for this problem: We will most probably change our Introspection so that `cuid()` and `uuid()` will stay in the schema. As we have no way to pick these up from the database, that is the right thing to do anyway. Will take a bit longer, but we will post a link to an issue here soon.

**Comment by do4gr** [comment]
Quick update: once this is merged https://github.com/prisma/prisma-engines/pull/1030 you should be able to take your P1 schema, run the upgrade tool and then run `prisma introspect --experimental-reintrospection` and it should keep the `cuid()` and `uuid()` values. 

This is probably going to be part of Tuesdays release.

**Comment by do4gr** [comment]
We just merged the fix via re-introspection.

**Comment by matthewmueller** [comment]
I'll re-open as I need to bump the prisma-sdk after release.

**Comment by alan345** [comment]
Hello @matthewmueller. Is the Tuesday realesed is live?

**Comment by janpio** [comment]
Now it is: https://github.com/prisma/prisma/releases/tag/2.5.0 (although this is not explicitly mentioned in the release notes - `npx prisma introspect --experimental-reintrospection` should work for this.)

**Comment by alan345** [comment]
Great! thank you team!
![image](https://user-images.githubusercontent.com/15246526/90569778-4ee71380-e163-11ea-867d-32514e4b9bce.png)

**Comment by janpio** [comment]
Oh, that's great - but I actually see a bug there: It does not list the fields this affects! I'll open a new issue for this.
